[BUG] iceberge

**Describe the bug**
first seen in rapids_integration-scala213-pre_release-github run: 228

spark355,
```
src.main.python.iceberg_test.test_iceberg_fallback_not_unsafe_row[DATAGEN_SEED=1747903411, TZ=UTC, IGNORE_ORDER({'local': True}), ALLOW_NON_GPU(BatchScanExec)]
src.main.python.iceberg_test.test_iceberg_unsupported_formats[COALESCING-avro-[Long]][DATAGEN_SEED=1747903411, TZ=UTC]
...
=========== 108 failed, 48 warnings, 108 errors in 206.44s (0:03:26) ===========
```

```
[2025-05-22T08:43:56.277Z] 25/05/22 08:43:55 ERROR GpuOverrideUtil: Encountered an exception applying GPU overrides java.lang.ExceptionInInitializerError
[2025-05-22T08:43:56.277Z] java.lang.ExceptionInInitializerError
[2025-05-22T08:43:56.277Z] 	at com.nvidia.spark.rapids.GpuOverrides.$anonfun$applyWithContext$1(GpuOverrides.scala:4888)
[2025-05-22T08:43:56.277Z] 	at com.nvidia.spark.rapids.GpuOverrideUtil$.$anonfun$tryOverride$1(GpuOverrides.scala:4855)
[2025-05-22T08:43:56.277Z] 	at com.nvidia.spark.rapids.GpuOverrides.applyWithContext(GpuOverrides.scala:4911)
[2025-05-22T08:43:56.277Z] 	at com.nvidia.spark.rapids.GpuOverrides.apply(GpuOverrides.scala:4882)
[2025-05-22T08:43:56.277Z] 	at com.nvidia.spark.rapids.GpuOverrides.apply(GpuOverrides.scala:4878)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.ApplyColumnarRulesAndInsertTransitions.$anonfun$apply$1(Columnar.scala:535)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.ApplyColumnarRulesAndInsertTransitions.$anonfun$apply$1$adapted(Columnar.scala:535)
[2025-05-22T08:43:56.277Z] 	at scala.collection.immutable.List.foreach(List.scala:333)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.ApplyColumnarRulesAndInsertTransitions.apply(Columnar.scala:535)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.ApplyColumnarRulesAndInsertTransitions.apply(Columnar.scala:487)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.QueryExecution$.$anonfun$prepareForExecution$1(QueryExecution.scala:477)
[2025-05-22T08:43:56.277Z] 	at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
[2025-05-22T08:43:56.277Z] 	at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
[2025-05-22T08:43:56.277Z] 	at scala.collection.immutable.List.foldLeft(List.scala:79)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.QueryExecution$.prepareForExecution(QueryExecution.scala:476)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executedPlan$1(QueryExecution.scala:186)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:138)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$2(QueryExecution.scala:219)
[2025-05-22T08:43:56.277Z] 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:546)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:219)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:218)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.executedPlan$lzycompute(QueryExecution.scala:186)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.executedPlan(QueryExecution.scala:179)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.simpleString(QueryExecution.scala:238)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.org$apache$spark$sql$execution$QueryExecution$$explainString(QueryExecution.scala:284)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.QueryExecution.explainString(QueryExecution.scala:252)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:117)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4321)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:4146)
[2025-05-22T08:43:56.278Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2025-05-22T08:43:56.278Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2025-05-22T08:43:56.278Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-05-22T08:43:56.278Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2025-05-22T08:43:56.278Z] 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-05-22T08:43:56.278Z] 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-05-22T08:43:56.278Z] 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-05-22T08:43:56.278Z] 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-05-22T08:43:56.278Z] 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-05-22T08:43:56.278Z] 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-05-22T08:43:56.278Z] 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-05-22T08:43:56.278Z] 	at java.base/java.lang.Thread.run(Thread.java:840)
[2025-05-22T08:43:56.278Z] Caused by: java.lang.ClassNotFoundException: com.nvidia.spark.rapids.iceberg.IcebergProviderImpl
[2025-05-22T08:43:56.278Z] 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445)
[2025-05-22T08:43:56.278Z] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
[2025-05-22T08:43:56.278Z] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
[2025-05-22T08:43:56.278Z] 	at com.nvidia.spark.rapids.ShimReflectionUtils$.loadClass(ShimReflectionUtils.scala:30)
[2025-05-22T08:43:56.278Z] 	at com.nvidia.spark.rapids.ShimReflectionUtils$.newInstanceOf(ShimReflectionUtils.scala:34)
[2025-05-22T08:43:56.278Z] 	at com.nvidia.spark.rapids.ShimLoaderTemp$.newIcebergProvider(ShimLoaderTemp.scala:56)
[2025-05-22T08:43:56.278Z] 	at com.nvidia.spark.rapids.iceberg.IcebergProvider$.apply(IcebergProvider.scala:29)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.rapids.ExternalSource$.icebergProvider$lzycompute(ExternalSource.scala:64)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.rapids.ExternalSource$.icebergProvider(ExternalSource.scala:64)
[2025-05-22T08:43:56.278Z] 	at org.apache.spark.sql.rapids.ExternalSource$.getScans(ExternalSource.scala:123)
[2025-05-22T08:43:56.278Z] 	at com.nvidia.spark.rapids.GpuOverrides$.<clinit>(GpuOverrides.scala:4170)
[2025-05-22T08:43:56.278Z] 	... 46 more
```

**Steps/Code to reproduce bug**
https://github.com/NVIDIA/spark-rapids/blob/branch-25.06/jenkins/spark-tests.sh#L215-L241

**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/iceberg/pom.xml
+++ b/iceberg/pom.xml
@@ -32,6 +32,7 @@
     <version>25.06.0-SNAPSHOT</version>
 
     <properties>
+        <rapids.module>iceberg</rapids.module>
         <rapids.compressed.artifact>false</rapids.compressed.artifact>
         <rapids.shim.jar.phase>package</rapids.shim.jar.phase>
         <maven.scaladoc.skip>true</maven.scaladoc.skip>

--- a/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/IcebergProviderImpl.scala
+++ b/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/IcebergProviderImpl.scala
@@ -30,7 +30,7 @@ class IcebergProviderImpl extends IcebergProvider {
     Seq(new ScanRule[Scan](
       (a, conf, p, r) => new ScanMeta[Scan](a, conf, p, r) {
         private lazy val convertedScan: Try[GpuSparkBatchQueryScan] = GpuSparkBatchQueryScan
-          .tryConvert(a, conf)
+          .tryConvert(a, this.conf)
 
         override def supportsRuntimeFilters: Boolean = true
 

--- a/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/reader.scala
+++ b/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/reader.scala
@@ -172,6 +172,7 @@ trait GpuIcebergParquetReader extends Iterator[ColumnarBatch] with AutoCloseable
             bloomFilter.shouldRead(typeWithIds, rowGroup, reader.getBloomFilterDataReader(rowGroup))
       }
     }.getOrElse(blocks)
+      .toSeq
   }
 
   def clipBlocksToSchema(fileReadSchema: ShadedMessageType,

--- a/iceberg/src/main/scala/org/apache/iceberg/spark/source/GpuSparkPartitioningAwareScan.scala
+++ b/iceberg/src/main/scala/org/apache/iceberg/spark/source/GpuSparkPartitioningAwareScan.scala
@@ -36,5 +36,5 @@ abstract class GpuSparkPartitioningAwareScan[T <: PartitionScanTask](
 
   override def groupingKeyType(): Types.StructType = cpuScan.groupingKeyType()
 
-  override def taskGroups(): Seq[_ <: ScanTaskGroup[_]] = cpuScan.taskGroups().asScala
+  override def taskGroups(): Seq[_ <: ScanTaskGroup[_]] = cpuScan.taskGroups().asScala.toSeq
 }

--- a/scala2.13/iceberg/pom.xml
+++ b/scala2.13/iceberg/pom.xml
@@ -32,6 +32,7 @@
     <version>25.06.0-SNAPSHOT</version>
 
     <properties>
+        <rapids.module>iceberg</rapids.module>
         <rapids.compressed.artifact>false</rapids.compressed.artifact>
         <rapids.shim.jar.phase>package</rapids.shim.jar.phase>
         <maven.scaladoc.skip>true</maven.scaladoc.skip>

